backend/control/control.go

```diff
@@ -34,130 +37,78 @@ type CartographyJobRequest struct {
-			resp.Body.Close()
+		if !active {
+			log.Printf("Rule is inactive. Skipping...")
 			continue
 		}
-		resp.Body.Close()
-		if sr.Status != "READY" {
+
+		// Execute rule
+		results, err := rules.ExecuteRule(driver, r)
+		if err != nil {
+			log.Printf("Error executing rule: %v", err)
 			continue
 		}
 
-		// Execute rules
-		for idx, r := range rulesToExecute {
-			log.Printf("Rule ID and description: %v %v", r.UID(), r.Description())
-
-			// Skip if rule is inactive
-			active, err := rules.IsRuleActive(postgresDb, r)
-			if err != nil {
-				log.Printf("Error querying for rule activity... %v", err)
-				continue
-			}
-			if !active {
-				log.Printf("Rule is inactive. Skipping...")
-				continue
-			}
-
-			// Execute rule
-			results, err := rules.ExecuteRule(driver, r)
-			if err != nil {
-				log.Printf("Error executing rule: %v", err)
-				continue
-			}
+		rd := ruleDataList[idx]
 
-			rd := ruleDataList[idx]
-
-			// Upsert rule results
-			err = rules.UpsertRuleResults(postgresDb, rd, results)
-			if err != nil {
-				log.Printf("Unexpected error upserting rule_results %v", err)
-				continue
-			}
-
-			// Delete stale rule results
-			err = rules.DeleteStaleRuleResults(postgresDb, rd)
-			if err != nil {
-				log.Printf("Unexpected error deleting unseen results %v", err)
-				continue
-			}
+		// Upsert rule results
+		err = rules.UpsertRuleResults(postgresDb, rd, results)
+		if err != nil {
+			log.Printf("Unexpected error upserting rule_results %v", err)
+			continue
 		}
 
-		// Set LastScanCompleted of every row in account_details
-		postgresDb.Model(&models.AccountDetails{}).Where("1 = 1").Update("last_scan_completed", time.Now())
-
-		// Reset ExecuteRules boolean
-		ExecuteRulesMutex.Lock()
-		ExecuteRules = false
-		ExecuteRulesMutex.Unlock()
+		// Delete stale rule results
+		err = rules.DeleteStaleRuleResults(postgresDb, rd)
+		if err != nil {
+			log.Printf("Unexpected error deleting unseen results %v", err)
+			continue
+		}
 	}
-}
 
-// TriggerScan attempts to
-//  1. Start a cartography job
-//  2. If the job was started or is already running, mark ExecuteRules boolean
-//     so the RuleExecutionLoop can run.
-func TriggerScan(postgresDb *gorm.DB) error {
-	// TODO: Read / parse credentials from database or request
-	var accountDetailsLst []models.AccountDetails
-	if tx := postgresDb.Find(&accountDetailsLst); tx.Error != nil {
+	tx := postgresDb.Model(&models.AccountDetails{}).Where("scan_status = ?", "RULES_RUNNING").Updates(map[string]interface{}{"scan_status": "PASSED", "last_scan_completed": time.Now()})
+	if tx.Error != nil {
 		return tx.Error
 	}
-	if len(accountDetailsLst) == 0 {
-		// No account details found, so skip cartography scanning
-		return nil
-	}
-	if len(accountDetailsLst) > 1 {
-		return fmt.Errorf("expected only 1 account details from db, got %v", len(accountDetailsLst))
-	}
+	return nil
+}
 
+func StartCartographyJob(account models.AccountDetails) error {
```

Comment:
add comment explaining what StartCartographyJob does and when it returns error

Reply:
there are many cases where it will return error, not practical to document all of them